src/python/grpcio_tests/tests/unit/test_common.py

```diff
@@ -132,3 +132,15 @@ def wait(self):
         while self.count > 0:
             self.cv.wait()
         self.cv.release()
+
+
+def running_under_gevent():
+    try:
+        from gevent import monkey
+        import gevent.socket
+    except ImportError:
+        return False
+    else:
+        import socket
+        is_gevent = socket.socket is gevent.socket.socket
+        return is_gevent
```

Comment:
nit: Gevent provides `monkey.is_module_patched`.

Reply:
Unfortunately, this version of `gevent` does not. I'm trying to use the same version of `gevent` that's been tested in our existing tests so that we don't change too many variables at once. According to some research, this method is the most backward-compatible.